### PR TITLE
feat(typescript): add d.ts to pacakge.json

### LINKED ISSUE #70: TypeScript: Add d.ts in npm
From #68

> This change only does rewrite codes with TypeScript. To publish d.ts in npm, we need push more patch onto this.

Can we use resolve this issue by using  babel `--copy-files` option?

- [ ] add `types` field in pacakge.json
- [ ] add *d.ts files to `lib/`

### PR DESCRIPTION
**Pull request by azu**
- Add `types` field to pacakge.json
- Add `clean` npm-run-script
- Use `cross-env` for setting eviroment

fix #70 

/cc @saneyuki

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,23 +19,25 @@
...
     "start:docs": "gitbook serve",
     "build:docs": "gitbook build",
-    "build": "npm-run-all -p build:tsc build:cp_obj -s build:down_transform",
+    "build": "npm-run-all -s clean build:src build:down_transform",
     "build:cp_obj": "cpx 'src/**/*.{js, jsx}' __obj/ --preserve",
     "build:tsc": "tsc --project .",
-    "build:down_transform": "NODE_ENV=production babel __obj --out-dir lib --source-maps",
-    "watch": "babel src --out-dir lib --watch --source-maps",
+    "build:src": "npm-run-all -p build:tsc build:cp_obj",
+    "build:down_transform": "cross-env NODE_ENV=production babel __obj --out-dir lib --source-maps",
```
Comment (saneyuki) [inline package.json]: I feel we should sort script names by the order of breaking down a problem.  So `build` -> `build:src` -> (`build:tsc` and `build:cp_obj`) -> `build:down_transform`

**Comment by azu** [comment]
> I seem the current patch does not have a part to copying d.ts files from /__obj/ to /lib/.

ah, I've forget `--copy-files` options for babel.
I'll fix it.

File: package.json (hunk 1 above)
Comment (azu) [inline package.json]: fixed https://github.com/almin/almin/pull/78/commits/d3d01c72095a4a618ce23a065db41bcc80b809da

**Comment by saneyuki** [comment]
> --copy-files options for babel.

I don't know well about its option. Is it handle source map files correctly?
I'm worry that babel does not emit the source map `__obj/`  -> `lib/` properly.

I feel it's more clear & robustness to use `cpx` to copy `d.ts` .

**Comment by azu** [comment]
> `--copy-files
> Copy files that will not be compiled
> -- https://babeljs.io/docs/usage/cli/

Compile files + Copy files that is not compiled.

**Comment by azu** [comment]
> I feel it's more clear & robustness to use cpx to copy d.ts .

ok.

**Comment by azu** [comment]
done
almin 2017-02-20 17-32-20

I've refactor npm run-script

- `build:src:*` :  building script for src/
- `build:lib:*` :  building script fro lib/

Finally, `npm run build` does  `npm-run-all -s clean build:src build:lib`.

**Comment by saneyuki** [comment]
looks good to me.

**Pull request merged**